server/test/unit/server/webservices/views/test_plugins.py

```diff
@@ -25,7 +26,49 @@ class TestImportersView(unittest.TestCase):
 
 
 class TestTypeResourceView(unittest.TestCase):
-    pass
+    """
+    Tests for the views for a single plugin type.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.plugins.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.plugins.factory')
+    def test_get_type_resource_view_existing_type(self, mock_factory, mock_serializer):
+        """
+        Type Resource should call the seralizer with the appropriate object.
+        """
+        mock_manager = mock.MagicMock()
+        mock_manager.types.return_value = [{'id': 'mock_type_1'}, {'id': 'mock_type_2'}]
+        mock_factory.plugin_manager.return_value = mock_manager
+        request = mock.MagicMock()
+        request.get_full_path.return_value = '/mock/path/'
+
+        type_resource = TypeResourceView()
+        type_resource.get(request, 'mock_type_2')
+
+        expected_content = {'id': 'mock_type_2', '_href': '/mock/path/'}
+        mock_serializer.assert_called_once_with(expected_content)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.plugins.factory')
+    def test_get_type_resource_view_nonexistent_type(self, mock_factory):
+        """
+        Type Resource should raise a MissingResource if the specified type does not exist.
+        """
+        mock_manager = mock.MagicMock()
+        mock_manager.types.return_value = [{'id': 'mock_type_1'}, {'id': 'mock_type_2'}]
+        mock_factory.plugin_manager.return_value = mock_manager
```

Comment:
This is actually not necessary. Since ``mock_factory`` is already a ``MagicMock``, attribute access and calls to __call__ will automatically create a new instance of ``MagicMock`` and assign it to the appropriate attribute or return_value.

```
>>> from mock import MagicMock
>>> m = MagicMock()
>>> m.a.b()
<MagicMock name='mock.a.b()' id='139911107642896'>
>>> m.a
<MagicMock name='mock.a' id='139911107592336'>
>>> m.a.b.return_value
<MagicMock name='mock.a.b()' id='139911107642896'>
```